### DISCUSSION TITLE
Need an option to add products along with their add-ons in the endpoint

### DISCUSSION THREAD

**Discussion by danishashraf047** [co-cart/co-cart#301] [Ideas]
CoCart is a great library and works well. There must be a feature to add products along with their add-ons.

I'm implementing your provided Restful APIs. Now, I'm stuck to adding products into the cart with their add-ons. I also have gone through the documentation but couldn't get any method of how we can add products with their add-ons. If you could add this feature so it would be good. I need to implement this feature in my client application.

If this feature is already presented in the existing version so kindly let us know how we can use it. I'm also providing a link where I got the same question. Thank you!

https://wordpress.org/support/topic/adding-product-with-product-add-ons-to-cart/

**Comment by danishashraf047** [comment]
Hi @seb86, Is there any option that already exists to do this, or you will add it?

**Reply by seb86** [reply]
@danishashraf047 I'm focused to keeping the core of CoCart for compatibility purposes either with authentication, cart identification, core WooCommerce compatibility, specific web hosts or frameworks.

Supporting big extensions will be more of a task for CoCart Pro to handle. As I already responded in that forum post, you can pass custom item data via the `item_data` parameter when you add an item to the cart.

It is on the roadmap to support the official WooCommerce Products Add-ons extension. What that will be I don't know yet. It maybe just a guide that I will provide. Will know more once v3.1 is made available next year.

**Comment by danishashraf047** [comment]
Alright, Thank you so much. :)

**Comment by danishashraf047** [comment]
I can pass product addons with item_data property but the price of the product is not increasing. When I pass nyp so it changes the product price. :(

**Reply by seb86** [reply]
@danishashraf047 Sorry for the late response. You need to have the Name Your Price extension available on WooCommerce.com installed and active for it to take affect.

CoCart 3.1 will have it's own built in price change which is coming out soon.
